### ISSUE TITLE
[java] UselessPureMethodCall Misses Discarded anyMatch Results on Primitive Streams

### ISSUE THREAD

**Issue by Emilyaxe** [pmd/pmd#6649]
**Affects PMD Version:** 7.24.0

**Rule:** UselessPureMethodCall — https://docs.pmd-code.org/latest/pmd_rules_java_errorprone.html#uselesspuremethodcall

**Description:**

`UselessPureMethodCall` correctly flags discarded `anyMatch(...)` results on `Stream<T>`, but fails to report the same pattern on primitive streams (`IntStream`, `LongStream`, `DoubleStream`). Since `anyMatch` is equally pure across both APIs and the returned `boolean` is unused in both cases, the missing report on primitive streams is a false negative.

**Code Sample demonstrating the issue:**

```java
import java.util.stream.DoubleStream;
import java.util.stream.IntStream;
import java.util.stream.LongStream;
import java.util.stream.Stream;

public class PmdPrimitiveStreamReproducer {

    // BEFORE: correctly flagged by UselessPureMethodCall
    public void boxedStreams() {
        Stream<Integer> iStream = Stream.of(1).filter(i -> i < 5);
        iStream.anyMatch(i -> i < 5);               // reported

        Stream<Long> lStream = Stream.of(1L).filter(i -> i < 5L);
        lStream.anyMatch(i -> i < 5L);              // reported

        Stream<Double> dStream = Stream.of(1.0).filter(i -> i < 5.0);
        dStream.anyMatch(i -> i < 5.0);             // reported
    }

    // AFTER: NOT flagged, although the smell is identical
    public void primitiveStreams() {
        IntStream iStream = IntStream.of(1).filter(i -> i < 5);
        iStream.anyMatch(i -> i < 5);               // NOT reported

        LongStream lStream = LongStream.of(1L).filter(i -> i < 5L);
        lStream.anyMatch(i -> i < 5L);              // NOT reported

        DoubleStream dStream = DoubleStream.of(1.0).filter(i -> i < 5.0);
        dStream.anyMatch(i -> i < 5.0);             // NOT reported
    }
}
```

**Expected outcome:** `UselessPureMethodCall` is reported on each `IntStream.anyMatch`, `LongStream.anyMatch`, and `DoubleStream.anyMatch` call whose result is discarded, matching the behavior on `Stream<T>.anyMatch`.

**Actual outcome:** Only the `Stream<T>.anyMatch` calls are flagged; the three primitive-stream variants produce no violation.

**Running PMD through:** CLI

**Comment by UncleOwen** [comment]
Thanks for reporting. I'm closing this, because it's a duplicate. The good news: It's already fixed and will be in the next version!
